### ISSUE TITLE
[REVIEW]: Differential geometric algebra using Leibniz, Grassmann

### ISSUE THREAD

**Issue by whedon** [JuliaCon/proceedings-review#18]
**Submitting author:** @chakravala (<a href="http://orcid.org/0000-0002-8006-7816">Michael Reed</a>)
**Repository:** <a href="https://github.com/chakravala/Grassmann.jl" target ="_blank">https://github.com/chakravala/Grassmann.jl</a>
**Editor:** @matbesancon
**Reviewer:** @eschnett, @snoeyink
**Archive:** Pending

## Status

[![status](https://proceedings.juliacon.org/papers/17bff9097fc61aa1fd11aee4dae835b8/status.svg)](https://proceedings.juliacon.org/papers/17bff9097fc61aa1fd11aee4dae835b8)

Status badge code:

```
HTML: <a href="https://submissions.juliacon.org/papers/17bff9097fc61aa1fd11aee4dae835b8"><img src="https://submissions.juliacon.org/papers/17bff9097fc61aa1fd11aee4dae835b8/status.svg"></a>
Markdown: [![status](https://submissions.juliacon.org/papers/17bff9097fc61aa1fd11aee4dae835b8/status.svg)](https://submissions.juliacon.org/papers/17bff9097fc61aa1fd11aee4dae835b8)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/chakravala/Grassmann.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@eschnett & @snoeyink, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/JuliaCon/proceedings-reviews/invitations

The reviewer guidelines are available here: https://proceedings.juliacon.org/guide/reviewers. Any questions/concerns please let @matbesancon know.

✨ **Please try and complete your review in the next two weeks** ✨

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/chakravala/Grassmann.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Authorship:** Has the submitting author (@chakravala) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [ ] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [ ] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [ ] **Context:** is the scientific context motivating the work correctly presented?
- [ ] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [ ] **Results:** are the results presented and compared to approaches with similar goals?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/chakravala/Grassmann.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Authorship:** Has the submitting author (@chakravala) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [x] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [ ] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [ ] **Context:** is the scientific context motivating the work correctly presented?
- [ ] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [ ] **Results:** are the results presented and compared to approaches with similar goals?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @eschnett, @snoeyink it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/JuliaCon/proceedings-review) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/JuliaCon/proceedings-review:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #18 with the following error: 

 rm: cannot remove '*.aux': No such file or directory
Latexmk: This is Latexmk, John Collins, 17 Jan. 2018, version: 4.55.
Rule 'pdflatex': Rules & subrules not known to be previously run:
   pdflatex
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'pdflatex  -recorder  "paper.tex"'
------------
===========Latexmk: Missing input file: 'dirtytalk.sty' from line
  '! LaTeX Error: File `dirtytalk.sty' not found.'
Latexmk: Missing input file: 'dirtytalk.sty' from line
  '! LaTeX Error: File `dirtytalk.sty' not found.'
Failure to make 'paper.pdf'
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 1
      Refer to 'paper.log' for details
Looks like we failed to compile the PDF

**Comment by matbesancon** [comment]
@chakravala there are still issues in the paper folder

**Comment by eschnett** [comment]
@chakravala I used `lualatex` and `bibtex` to format the paper. There are problems with the references:
- `bibtex` outputs an error message about commas in the author list
- Grassmann didn't publish in 2000
- several references are incomplete, e.g. urls might be missing
- the name "de Rham" appears in lower case; this is probably bibtex's fault that quite eagerly insists that titles should be all lower case

**Comment by eschnett** [comment]
@chakravala The manuscript itself is very interesting, although I haven't read it in detail yet. However, I notice that it is quite technical and doesn't fit with the target audience at JuliaCon. At least, an introduction to geometric algebra is missing, and a description why Julia is a good implementation language. It seems to me that mentioning the geometric aspects (rotations, reflections) and the connection to computer visualization in the introduction might go well with the Julia crowd.

I also notice that the paper is quite terse. For the record: neither the reviewer nor the author instructions contain a page limit.

**Comment by chakravala** [comment]
> there are still issues in the paper folder

> bibtex outputs an error message about commas in the author list

I will have a look at resolving that

> Grassmann didn't publish in 2000

The translation from the AMS was published in 2000

> neither the reviewer nor the author instructions contain a page limit.

I thought there was a limit for 6 pages, so I had to leave out details I would otherwise want to include, can you confirm that the paper can really be longer than 6 pages?

Is it possible to make the paper 1-column, or should I convert to 2-columns?

**Comment by snoeyink** [comment]
Summary of Comments on grassmann-juliacon-2019.pdf (I'll upload a commented pdf.)
Page: 1

This paper is a description of Grassmann.jl and its supporting packages, which are appear quite impressive and extensive. Unfortunately, it
assumes extensive knowledge of terminology and notation from the reader, without ever defining what that reader should know, whose
notations they should be familiar with, and what that reader can expect to gain from reading.

As someone who has written a lot of geometric and topological code in low dimensions, and been interested in Clifford algebra enough
to read the book of Dorst, Fontijne, & Mann and some of Hestenes, I was eager to review this. I was disappointed because it does not
clearly say what the package does, nor what decisions were made between alternative implementations. (There are many intriguing hints,
e.g., Due to the design of the VectorBundle type
system, these operations make code optimization easier at compile-time by evaluating the bit parameters.)

As Fred Brooks has said, "Since anyone can create something new [in a synthetic field], that alone does not establish a contribution.
Rather, one must show that the creation is better." (Quoted from https://cra.org/resources/best-practice-memos/evaluating-computerscientists-
and-engineers-for-promotion-and-tenure/) A paper should do more than quote from the documentation of a package; it
should provide context so the reader (from the author's desired audience) can understand the decisions made by the package author and
why they are better than alternatives.

I feel that I could learn from this package, but not the document, as the author is working entirely from his own context, and does not
attempt to reach a broader audience. Unfortunately, the GPL3 license also discourages me from exploring the package source because I can't
guarantee that I can GPL3 any results that I mey eventually derive.

There are many easy things that can be done to broaden the audience for the paper.

1. Ask yourself, "Who is the intended audience? What is their expected background knowledge and what can they expect to learn from reading
this and can expect to get from the package?" Then answer these questions in the abstract or introduction.

2. Focus on verbs.
Keep action in the present tense (e.g., defn 4 & after).
For package descriptions or section headings, a verb would help parse: (
DirectSum.jl: provides types and operations for abstract tangent bundle vector space (unions, intersections, sums, etc.)
1 Parametric type polymorphism supports DirectSum )
If half the verbs in a paragraph are passive or being verbs (is, are, was), consider rewriting.

3. Strengthen parallels: use consistent terms (factorize = partition?
Many places compare and contrast the Leibniz and Grassmann algebras,, the mathematics and its implementation, the type hierarchy and
executable code. Write about parallel concepts using the same wording and sentence structure so that the similarities and differences pop out.

4. Define before using. Defn 1 defines its notation (except for n) after using it, forcing the reader to move back and forth. (In fact, it is unclear if
defn 1 is defining a type or a data representation.) Define vector space V\in Vect_K with metric g:VxV-->K,
Defn 5 defines almost none of the notation it uses.

5. Context first. (See Gopen & Swan https://cseweb.ucsd.edu/~swanson/papers/science-of-writing.pdf, or Gopen's book "The Sense of Structure")
Gopen says that a reader of English expects context first, new information after, in a sentence, paragraph, or section. By providing the
information in this order you conserve the reader energy for understanding the idea

How should section heading 1 be parsed?

Page: 3

is denoted by
[Stay in present tense, and improve parallelism.]


This is defining two different notations for symmetric difference.

This paragraph on implementation of the mathematics defined above should more clearly parallel the mathematics and should indicate
what it means to "work with" a TensorAlgebra{V}. Rewrite context first, and focus on verbs. You also have the parallel between
TensorAlgebra{V} and Grassman.Algebra{V}: structure the discussions the same, so that the differences in the concepts are not masked by
differences in expression.

Here is what I might say to restructure this paragraph in implementation, starting with motivation:

The bases for Grassmann and Leibniz algebras are large, since they include all sets or multiset of basis elements. Dorst, Fontijne, & Mann
suggest that the efficient implementation of geometric algebra is a compiler problem, to be solved by caching the product computations
actually needed in an application. Julia's parametric types and multiple dispatch are well suited for this. TensorAlgebra{V} is designed so
a user can smoothly transition between very high dimensional and low dimensional algebras in a single session, with varying levels of caching
and optimization. Printing with full alpha-numeric labeling with lower-case and capital letters limits a Basis to $n\le 62$ indices. Memory limits a
full MultiVector to n ≤ 22, although sparse operations support higher dimensions. A Grassmann.Algebra{V} is a container for the TensorAlgebra
generated by V, a VectorBundle{n}. Grassmann.Algebra{V} supports full caching for n ≤ 8, Grassmann.SparseAlgebra supports dimensions n ≤ 22, and
Grassmann.ExtendedAlgebra supports higher dimensions.


Consider rewriting the first paragraph of section 3 "context first":
I think this paragraph is attempting to say that both Grassmann's exterior algebra and Leibnitz's algebra of symmetric differential functions are
both captured by the geometric product. (Some of it is a repeat of the concepts below Defn 4.)
----original, with comment----
While oriented sets of the Grassmann exterior algebra are simpler,
[simpler than what? The reader has no context yet. Starting with "while" (and, below, "when") misleads your reader into thinking that time is
important to the context: while --> although.]
the parity [1] of (−1)Π is factored into
[The reader has to parse [1] as a citation and (-1)Π as notation.]
transposition compositions when interchanging the ordering of the tensor product argument permutation.
[Context is at the end of this sentence, and this is really hard to parse -- interchanging applies to two things, argument is singular, permutation is
collective.
Might one say, "Transposing terms in a tensor product changes the parity in (-1)Π."]
Alternatively, symmetric algebra does not need to track the parity
[By using "track the parity" to describe the process of the previous sentence, you force the reader to reinterpret what they just read.]
but does have multiplicity in its indices. [First use of indices -- would be a useful to give context for transpositions for tensor product...]
Symmetric differential function algebra of Leibniz trivializes the orientation property of index multi-sets, ["trivializes the orientation property" =
"makes tracking orientation trivial?" ]
while Grassmann’s exterior algebra is partitioned
into two oriented equivalence classes by anti-symmetry. The full tensor algebra [is this the symmetric differential function algebra?]
can be [sub-partitioned --> partitioned -- is this the same as the "can be factored" above?] into equivalence classes in multiple ways based on the composite symmetry, grade, and metric signature properties.
Both of the symmetry classes [This is the first reference to symmetry classes. Does this refer to both equivalence classes? ]
can be characterized by
the same geometric product, which is written as multiplication but explicitly denoted by ⊖ for clarity here.

---Context first rewrite--- (see prev comment)
Grassmann's exterior algebra and Leibnitz's algebra of symmetric differential functions track orientation differently.
Exterior algebra partitions space into two classes equivalence classes by orientation; transposing terms in a tensor product changes the parity in (-1)Π.

Symmetric algebra includes terms with multiplicity. Rather than tracking parity, it has multiple options to partition into equivalence classes based
on composite symmetry, grade, and metric signature properties.
Both algebras can be expressed by the same geometric product, which is usually written implicitly as multiplication but here explicitly denoted by ⊖.

-- I realize that this is probably not what was intended, as it does not capture the anti-symmetric and symmetric index sets, but to capture that the reader needs more context to be introduced.

There is no clarity when you define \ominus in terms of undefined \omega_X and \eta_Y.

This expression is quite useless, since there are so many undefined terms: \PI and g are mentioned before, with \oplus and \otimes, but
\omega and \eta never appear on the right; w, i, \epsilon, \otimes \mu do, but their relationship to \omega and \eta is not defined. I find
it wrong to use \ominus for set symmetric difference in a definition of \ominus for geometric product.

Page 6
GPL 3 does inhibit extension and learning from this package in the minds of some.

The bibliographic references are incomplete. Include publisher and DOIs so that people can find what you are citing. If you, for example,
cite the Dover reprinting of Emil Artin's 1957 Geometric Algebra, include also the original publication date. http://www.cws.illinois.edu/workshop/writers/citation/apa/reprintedbook/ 

Put braces [ ] around to preserve capitalization of names in bibtex.

**Comment by snoeyink** [comment]
I've created an issue and uploaded an annotated pdf in: https://github.com/chakravala/Grassmann.jl/issues/29

This paper is a description of Grassmann.jl and its supporting packages, which are appear quite impressive and extensive. Unfortunately, it
assumes extensive knowledge of terminology and notation from the reader, without ever defining what that reader should know, whose
notations they should be familiar with, and what that reader can expect to gain from reading.

**Comment by vchuravy** [comment]
> can you confirm that the paper can really be longer than 6 pages?

Yes a paper can be longer than 6 pages, we do not have an explicit page-limit but we expect most papers to be 6-8 pages, longer if necessary.

**Comment by matbesancon** [comment]
ping @chakravala, could you make the modifications following the reviewers' comments?

**Comment by chakravala** [comment]
@matbesancon yes, I will finish it up, I have been traveling for the past month, but just got back and am going to finish this up within the next days.

**Comment by matbesancon** [comment]
ping @chakravala to continue here

**Comment by chakravala** [comment]
is it too late to finish submitting this paper?

the main problem is, with the 2 columns I can't fit my big equations into the page.. is there a way to split an equation across both columns?

**Comment by matbesancon** [comment]
no it's fine but it would be nice to finish up soon.

> the main problem is, with the 2 columns I can't fit my big equations into the page.. is there a way to split an equation across both columns?

Not sure it's possible, I'll try to find something, in the mean time it would be best to split your equations

**Comment by chakravala** [comment]
unfortunately, the `juliacon` document class does not seem to function properly. in the example document it is claimed that there are `definition, example, remark` environments, however when I try to compile my document they cause compilation errors. is this going to be fixed in the `juliacon` class?

**Comment by matbesancon** [comment]
@chakravala indeed the names of the environments are incorrect, you can check them out in the cls file:

```
\newtheorem{theorem}{Theorem}
\newtheorem{strategy}{Strategy}
\newtheorem{property}{Property}
\newtheorem{proposition}{Proposition}
\newtheorem{lemma}[theorem]{Lemma}
\newtheorem{exam}{Example}
\newenvironment{example}{\italicenvfalse\begin{exam}}{\end{exam}\italicenvtrue}
\newtheorem{defi}[theorem]{Definition}
```

**Comment by chakravala** [comment]
@matbesancon this is what I needed to change to make it work https://github.com/chakravala/Grassmann.jl/commit/347fb907c0dd289d57c0c1ceb78a31a133be1402

```
\newtheorem{theorem}{Theorem}
\newtheorem{strategy}{Strategy}
\newtheorem{property}{Property}
\newtheorem{proposition}{Proposition}
\newtheorem{lemma}[theorem]{Lemma}
\newtheorem{exam}{Example}
\newenvironment{example}{\begin{exam}}{\end{exam}}
\newtheorem{defi}[theorem]{Definition}
\newenvironment{definition}{\begin{defi}}{\end{defi}}
\newtheorem{rem}{Remark}
\newenvironment{remark}{\begin{rem}}{\end{rem}}
```
the `italicenv` stuff didn't work at all

draft is updated, all equations now fit into 4 pages, will post another update soon

**Comment by chakravala** [comment]
Hi folks, sorry about being a bit late with my paper.

I have now implemented the finalizing feature for the differential geometric algebra, which will allow me to add more explicit and concrete examples with visualization into the paper.

Please be a little more patient, it should only take about 1 more week to wrap up this review, as it is only a matter of sorting out the some details for the examples.

**Comment by matbesancon** [comment]
> this is what I needed to change to make it work chakravala/Grassmann.jl@347fb90

I think you can introduce that in the paper and not in the cls file

**Comment by chakravala** [comment]
No, the cls file had to be edited because otherwise it causes a compilation error, specifically `italicenv` causes the compilation error and must be removed from the cls. I can move the extra definitions to the main paper, but the cls filed needs to have the `italicenv` part removed.

**Comment by matbesancon** [comment]
You should not change the cls overall, if the italicenv doesn't work there is an incompatible part in the paper

**Comment by chakravala** [comment]
I have committed the final draft for my JuliaCon paper, here is a PDF download of it

[![DropBox](https://img.shields.io/badge/download_PDF-DropBox-blue.svg)](https://www.dropbox.com/sh/tphh6anw0qwija4/AAACiaXig5djrLVAKLPFmGV-a/Geometric-Algebra?preview=grassmann-juliacon-2019.pdf)

**Comment by matbesancon** [comment]
Hi, the paper really need to be built with the same environment to be accepted for the proceedings of the JuliaCon, it basically ensures the layout and the tooling will work for all papers we publish

**Comment by vchuravy** [comment]
@whedon withdraw

**Comment by whedon** [comment]
Paper withdrawn.

**Comment by chakravala** [comment]
Why is it being with drawn?

**Comment by vchuravy** [comment]
I withdrew your paper since you did not reply to https://github.com/JuliaCon/proceedings-review/issues/18#issuecomment-550494959 without the paper in the right format it is impossible for us to proceed.

**Comment by chakravala** [comment]
Well, I prepared everything, except that I needed to modify the `cls` file a bit.

What am I supposed to do to address the issue? I submitted everything, the paper compiles fine.

I would like to have this published.

**Comment by chakravala** [comment]
Would be a bit lame if you cancel my paper, only because you can't be bothered to deal with the 3 lines of code I modified in the template in order to make my paper compile.

**Comment by matbesancon** [comment]
to come back to the issue of publishing it, we can't let you modify the template, because we only provide it to authors in order to be able to build locally and not rely on pushing changes to a github repo. This means modifying the template creates an out-of-sync with the actual template we used, and we need to keep papers standardized.

The way forward is to take the TeX and CLS template from https://github.com/JuliaCon/JuliaConSubmission.jl for the paper. Again, we don't have the time or manpower to restructure the content.

Given the comments of the reviewers and the current content as shared on Dropbox, I would argue that the paper is not adapted to JuliaCon as a publishing venue and out of scope. It can be of interest, but needs to make less assumption on knowledge of differential geometry.

A large part of the paper is setting definitions, definitions should be referred to in other papers, bringing a whole lot of new symbols and notation does not help the reader understand what the package is doing. Same thing for theorems.

**Comment by chakravala** [comment]
Thank you for the response.

> A large part of the paper is setting definitions, definitions should be referred to in other papers, bringing a whole lot of new symbols and notation does not help the reader understand what the package is doing. Same thing for theorems.

It is completely necessary to go over all the definitions in this paper, since some of the definitions are entirely new and have never been published before. So, I literally can't reference other papers for some of these definitions.

I am willing to add 2 more pages to the paper to expand more on what the package is doing. It is now 1 year later, and I have much more insight into how this package can be used and what it can be used for.

**Comment by snoeyink** [comment]
While I realize that one can't allow authors to edit latex templates that are used uniformly for all articles, I'll be sad if the resolution is that this does not get published at all.  
I've used tex since 1985 and latex since 1990, so I'd be happy to work with you to make something that would compile with the standard template. (I was just contacted as a technical reviewer, and have no pull within JuliaCon for whether something can still be accepted after it was dropped) 
Last semester was completely crazy, and this semester promises to be only a little better, but I'd still like to connect with you at some point and learn from your experiences with geometric algebra in Julia.

**Comment by matbesancon** [comment]
> It is completely necessary to go over all the definitions in this paper, since some of the definitions are entirely new and have never been published before.

Ok I see, in that case, as highlighted before, the novel contributions in differential algebraic geometry are probably more adapted to another paper submitted to a mathematics venue (journal or conference), with this proceeding highlighting more the package.

**Comment by matbesancon** [comment]
> I am willing to add 2 more pages to the paper to expand more on what the package is doing. It is now 1 year later, and I have much more insight into how this package can be used and what it can be used for.

I don't think the solution is to just add more pages, but restructure the current content and focus on what you propose here: possible applications of Grassmann.jl and how to use it with some examples

**Comment by eschnett** [comment]
Since this paper has been public for more than a year, I would suggest posting a preprint to arxiv.org to make it more official. Many physics, math, and computer science papers are effectively published there before they are submitted to a journal (all my papers are).

As for a description for a Julia package, another venue is JOSS <https://joss.theoj.org>. The publications there are essentially an extended documentation for a software package.

**Comment by matbesancon** [comment]
I agree for the preprint aspect.
Thing to note for JOSS, there will be the same scope issue since JOSS is not meant to publish new contribution in the field of the software library, but review the library itself

**Comment by chakravala** [comment]
> probably more adapted to another paper submitted to a mathematics venue (journal or conference), with this proceeding highlighting more the package.

Indeed, I am currently writing a paper to a mathematics journal (Advances in Applied Clifford Algebras), but at the original time of writing of the JuliaCon paper that wasn't the case yet, since I only submitted that abstract recently.

I am also currently submitting a paper to American Institute of Aeronautics and Astronautics on applications of this package.

> I would suggest posting a preprint to arxiv.org to make it more official.

That's not possible, since I do not have permission to post on arxiv.

> JOSS is not meant to publish new contribution in the field of the software library, but review the library itself

It may be possible for me to modify the JuliaCon paper to meet this criteria, since I am also now submitting 2 other papers, which I could reference in the JuliaCon paper. At the original time of writing, that wasn't the case yet.

However, if it requires a complete rewrite like this, then I may not be able to fix the paper immediately in a short period of time.

**Comment by matbesancon** [comment]
As general advice, if you are writing and submitting multiple papers at the same time, having a preprint available (the version of the paper you submitted) is generally a good way to reference some of them in the others. If not arxiv, then a personal website is an okay choice, but an established preprint server will be preferred by most venues.

If you are re-structuring some parts in another article and citing it, then yes this will likely need some rewrite here as to not repeat what you already cite

**Comment by eschnett** [comment]
> That's not possible, since I do not have permission to post on arxiv.
Can you explain? arxiv.org is quite general in accepting authors and submissions.

**Comment by chakravala** [comment]
Have you tried submitting a paper on arxiv before? I tried it, and found that I can't just make an account and submit things.

**Comment by eschnett** [comment]
Yes, I have; see <http://arxiv.org/a/schnetter_e_1>. I find that arxiv.org works quite well. If there's a specific thing holding you up, then you should email their help desk.

**Comment by chakravala** [comment]
Honestly, i dont even care about posting my articles on arxiv or some journal, I'm just doing it for fun and couldn't care less how official anyone thinks it is. I have written hundreds of emails criticising univerisities to the point that I will never be able to get a job. I'll think about rewriting this paper, but actually I dont care what anyone out there thinks about my work anymore, you're all planning to leech off my work for free while I worked my ass off for years learning this, so I am not going to bend over backwards to please you or your publication, but I will casually work on rewriting some more if i feel like it, but honestly I dont know why I should bother with making my hard work available on someone else's preprint server, especially if those academic institutions sanctioned me and want me to suffer the fate of not having economic opportunities.

**Comment by chakravala** [comment]
Apologies if I offended anyone with my last (now deleted) post, it's not directed at anyone specifically, it was just me being abstractly frustrated.
